src/kixi/heimdall/handler.clj

```diff
@@ -52,14 +57,13 @@
            (re-find (re-pattern (str "^" token-name " (.+)$")))
            (second)))
 
+
 (defn create-group [req]
-  (let [[ok? res] (service/create-group (:cassandra-session (:components req))
-                                        (:auth-conf req)
-                                        (:user req)
-                                        (:params req))]
+  (let [ok? (and (spec/valid? :kixi.heimdall.schema/group-params (:params req)) (spec/valid? :kixi.heimdall.schema/user (:user req)))]
     (if ok?
-      {:status 201 :body res}
-      {:status 401 :body res})))
+      (do (comms/send-event! (:communications (:components req)) :kixi.heimdall/group-created "1.0.0" (merge {:group (:params req)} {:user (:user req)}))
```

Comment:
Might be worth putting the `:communications` extraction into a fn as it happens a lot